cloud/docker.go

```diff
@@ -249,3 +249,78 @@ func (m *dockerManager) GetContainers(ctx context.Context, h *host.Host) ([]stri
 
 	return ids, nil
 }
+
+// GetContainersRunningImage returns all the containers that are running a particular image
+func (m *dockerManager) GetContainersRunningImage(ctx context.Context, h *host.Host, imageID string) ([]string, error) {
+	containers, err := m.GetContainers(ctx, h)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error listing containers")
+	}
+	containersRunningImage := make([]string, 0)
+	for _, containerID := range containers {
+		container, err := m.client.GetContainer(ctx, h, containerID)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Error getting information for container '%s'", containerID)
+		}
+		if container.Image == imageID {
+			containersRunningImage = append(containersRunningImage, containerID)
+		}
+	}
+	return containersRunningImage, nil
+}
+
+// RemoveOldestImageID finds the oldest image and forcibly removes it
+func (m *dockerManager) RemoveOldestImageID(ctx context.Context, h *host.Host) error {
+	images, err := m.client.ListImages(ctx, h)
+	if err != nil {
+		return errors.Wrap(err, "Error listing images")
+	}
+	// get oldest image ID
+	oldest := time.Now()
+	id := ""
+	for _, image := range images {
+		if time.Unix(image.Created, 0).Before(oldest) {
+			oldest = time.Unix(image.Created, 0)
+			id = image.ID
+		}
+	}
+
+	if id == "" {
+		return errors.Errorf("No images to delete on host '%s'", h.Id)
+	}
+
+	// remove containers on image before removing the image
+	containersRunningImage, err := m.GetContainersRunningImage(ctx, h, id)
+	if err != nil {
+		return errors.Wrapf(err, "Error getting containers running on image '%s'", id)
+	}
+
+	if len(containersRunningImage) > 0 {
+		for _, containerID := range containersRunningImage {
+			if err := m.client.RemoveContainer(ctx, h, containerID); err != nil {
```

Comment:
Instead of removing the container, let's continue onto the next oldest

Reply:
Done.